### PR TITLE
Add Symbool support in python to C++ translation

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #98879
* #98878
* #98877
* __->__ #98453

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: aten/src/ATen/core/ivalue.cpp [hunk 1]
```diff
@@ -309,9 +311,9 @@ IValue IValue::equals(const IValue& rhs) const {
     case Tag::SymInt:
       return rhs.isSymInt() && lhs.toSymInt() == rhs.toSymInt();
     case Tag::SymFloat:
-      // NB: this doesn't actually work as sym floats don't have equality
```
Comment (tugsbayasgalan) [inline aten/src/ATen/core/ivalue.cpp:312]: outdated comment

File: torch/csrc/jit/python/pybind_utils.cpp [hunk 2]
```diff
@@ -451,6 +474,8 @@ IValue toIValue(py::handle obj, const TypePtr& type, c10::optional<int32_t> N) {
         return py::cast<c10::SymInt>(obj);
       } else if (torch::is_symfloat(obj)) {
         return py::cast<c10::SymFloat>(obj);
+      } else if (torch::is_symbool(obj)) {
+        return py::handle(obj).cast<c10::SymBool>();
```
Comment (ezyang) [inline torch/csrc/jit/python/pybind_utils.cpp:478]: Weirdly asymmetric

**Review by ezyang — APPROVED** [review]
Oof thanks for seeing this all the way through! I would recommend a teeny test or two

File: torch/csrc/jit/python/pybind_utils.cpp (hunk 2 above)
Comment (tugsbayasgalan) [reply torch/csrc/jit/python/pybind_utils.cpp:478]: Oh yes this is not needed.

**Pull request closed without merging**